specification/keyvault/data-plane/TypeSpec/models.tsp

```diff
@@ -0,0 +1,962 @@
+  string,
+  EC: "EC",
+  EC_HSM: "EC-HSM",
+  RSA: "RSA",
+  RSA_HSM: "RSA-HSM",
+  oct: "oct",
+  oct_HSM: "oct-HSM",
+}
+
+/**
+ * Elliptic curve name. For valid values, see JsonWebKeyCurveName.
+ */
+union JsonWebKeyCurveName {
+  string,
+  P_256: "P-256",
+  P_384: "P-384",
+  P_521: "P-521",
+  P_256K: "P-256K",
+}
+
+union KeyUsageType {
+  string,
+  digitalSignature: "digitalSignature",
+  nonRepudiation: "nonRepudiation",
+  keyEncipherment: "keyEncipherment",
+  dataEncipherment: "dataEncipherment",
+  keyAgreement: "keyAgreement",
+  keyCertSign: "keyCertSign",
+  cRLSign: "cRLSign",
+  encipherOnly: "encipherOnly",
+  decipherOnly: "decipherOnly",
+}
+
+/**
+ * The certificate list result.
+ */
+model CertificateListResult is Azure.Core.Page<CertificateItem>;
+
+/**
+ * The certificate item containing certificate metadata.
+ */
+model CertificateItem {
+  /**
+   * Certificate identifier.
+   */
+  id?: string;
+
+  /**
+   * The certificate management attributes.
+   */
+  attributes?: CertificateAttributes;
+
+  /**
+   * Application specific metadata in the form of key-value pairs.
+   */
+  tags?: Record<string>;
+
+  /**
+   * Thumbprint of the certificate.
+   */
+  @encodedName("application/json", "x5t")
+  @encode("base64url")
+  x509Thumbprint?: bytes;
+}
+
+/**
+ * The certificate management attributes.
+ */
+model CertificateAttributes extends Attributes {
+  /**
+   * softDelete data retention days. Value should be >=7 and <=90 when softDelete enabled, otherwise 0.
+   */
+  @visibility("read")
+  recoverableDays?: int32;
+
+  /**
+   * Reflects the deletion recovery level currently in effect for certificates in the current vault. If it contains 'Purgeable', the certificate can be permanently deleted by a privileged user; otherwise, only the system can purge the certificate, at the end of the retention interval.
+   */
+  @visibility("read")
+  recoveryLevel?: DeletionRecoveryLevel;
+}
+
+/**
+ * The object attributes managed by the KeyVault service.
+ */
+model Attributes {
+  /**
+   * Determines whether the object is enabled.
+   */
+  enabled?: boolean;
+
+  /**
+   * Not before date in UTC.
+   */
+  @encodedName("application/json", "nbf")
+  @encode("unixTimestamp", int32)
+  notBefore?: utcDateTime;
+
+  /**
+   * Expiry date in UTC.
+   */
+  @encodedName("application/json", "exp")
+  @encode("unixTimestamp", int32)
+  expires?: utcDateTime;
+
+  /**
+   * Creation time in UTC.
+   */
+  @visibility("read")
+  @encode("unixTimestamp", int32)
+  created?: utcDateTime;
+
+  /**
+   * Last updated time in UTC.
+   */
+  @visibility("read")
+  @encode("unixTimestamp", int32)
+  updated?: utcDateTime;
+}
+
+/**
+ * The key vault error exception.
+ */
+@error
+model KeyVaultError {
+  /**
+   * The key vault server error.
+   */
+  @visibility("read")
+  error?: Error;
+}
+
+/**
+ * The key vault server error.
+ */
+model Error {
+  /**
+   * The error code.
+   */
+  @visibility("read")
+  code?: string;
```

Comment:
Not related to the converter: can we use the Azure.Core.Foundations.ErrorResponse or Error type here?